### PR TITLE
lock and unlock documents

### PR DESCRIPTION
**Pull request by MicLi01**

### CODE REVIEW

File: server/database/models/doc.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ const DocSchema = new Schema({
   content: {type: String, default:''},
   original_content: {type: String, default:''},
   owner_id: {type: Schema.Types.ObjectId, ref: 'User'},
-  locked: Boolean,
+  locked: {type: Schema.Types.ObjectId, default: null},
```
Comment (melvinheks) [inline server/database/models/doc.js]: ref User

File: server/controllers/docHandlers.js [hunk 2]
```diff
@@ -51,6 +51,34 @@ const retrieveDoc = (req, res) => {
   });
 };
 
+const lockDoc = (req, res) => {
+  Doc.findOne({_id:req.params.documentId}, function(lockerr, lockresult){
```
Comment (melvinheks) [inline server/controllers/docHandlers.js]: lockErr, lockResult fix for unlockDoc as well

File: server/controllers/docHandlers.js [hunk 3]
```diff
@@ -51,6 +51,34 @@ const retrieveDoc = (req, res) => {
...
+    if(lockresult.locked) {
+      return res.status(404).json({msg:"Another user is currently making changes to the document"});
```
Comment (melvinheks) [inline server/controllers/docHandlers.js]: status(403).json({error

File: server/controllers/docHandlers.js [hunk 4]
```diff
@@ -51,6 +51,34 @@ const retrieveDoc = (req, res) => {
   });
 };
 
+const lockDoc = (req, res) => {
```
Comment (melvinheks) [inline server/controllers/docHandlers.js]: do login check

File: server/controllers/docHandlers.js [hunk 5]
```diff
@@ -51,6 +51,34 @@ const retrieveDoc = (req, res) => {
...
+      if(err) {
+        return res.status(500).json({error:"Failed to lock document"});
+      }
+      res.status(200).json({msg:"Locked document successfully"});
+    });
+  });
+}
+
+const unlockDoc = (req, res) => {
+  Doc.findOne({_id:req.params.documentId}, function(lockerr, lockresult){
```
Comment (melvinheks) [inline server/controllers/docHandlers.js]: lockErr, lockResult

File: server/controllers/docHandlers.js [hunk 6]
```diff
@@ -51,6 +51,34 @@ const retrieveDoc = (req, res) => {
...
+    Doc.updateOne({_id:req.params.documentId}, {locked: req.user._id}, function(err, result){
+      if(err) {
+        return res.status(500).json({error:"Failed to lock document"});
+      }
+      res.status(200).json({msg:"Locked document successfully"});
+    });
+  });
+}
+
+const unlockDoc = (req, res) => {
```
Comment (melvinheks) [inline server/controllers/docHandlers.js]: login check

File: server/controllers/docHandlers.js [hunk 7]
```diff
@@ -51,6 +51,34 @@ const retrieveDoc = (req, res) => {
...
+  Doc.findOne({_id:req.params.documentId}, function(lockerr, lockresult){
+    if(lockresult.locked != req.user._id && lockresult.owner_id != req.user._id) {
+      return res.status(500).json({msg:"You are not the user that currently has the lock"});
```
Comment (melvinheks) [inline server/controllers/docHandlers.js]: status(403).json({error

File: server/database/models/doc.js (hunk 1 above)
Comment (MicLi01) [reply server/database/models/doc.js]: ok

File: server/controllers/docHandlers.js (hunk 2 above)
Comment (MicLi01) [reply server/controllers/docHandlers.js]: ok

File: server/controllers/docHandlers.js (hunk 3 above)
Comment (MicLi01) [reply server/controllers/docHandlers.js]: ok

File: server/controllers/docHandlers.js (hunk 5 above)
Comment (MicLi01) [reply server/controllers/docHandlers.js]: ok

File: server/controllers/docHandlers.js (hunk 7 above)
Comment (MicLi01) [reply server/controllers/docHandlers.js]: ok

File: server/controllers/docHandlers.js (hunk 4 above)
Comment (MicLi01) [reply server/controllers/docHandlers.js]: ok

File: server/controllers/docHandlers.js (hunk 6 above)
Comment (MicLi01) [reply server/controllers/docHandlers.js]: ok

**Pull request closed without merging**